src/inference/src/dev/core_impl.cpp

```diff
@@ -1289,7 +1289,7 @@ void ov::CoreImpl::set_property_for_device(const ov::AnyMap& configMap, const st
                 if (device_supports_cache_dir(plugin.second)) {
                     ov::AnyMap empty_map;
                     auto cacheConfig = coreConfig.get_cache_config_for_device(plugin.second, empty_map);
-                    if (cacheConfig._cacheManager) {
+                    if (cacheConfig._cacheManager || cacheConfig._unsetCacheDir) {
                         configCopy[ov::cache_dir.name()] = cacheConfig._cacheDir;
```

Comment:
maybe we just need to remove `if (cacheConfig._cacheManager)` ?
In this case we always pass value of `cacheConfig._cacheDir` to underlying plugins.

Reply:
ping

Reply:
I think so, because:
1) empty dir: _cacheManager=null, _unsetCacheDir=true
2) non empty dir: _cacheManager=non_null, _unsetCacheDir=false

So `if (cacheConfig._cacheManager || cacheConfig._unsetCacheDir)` always be true.